app/services/user_with_asserted_attributes.rb

```diff
@@ -0,0 +1,18 @@
+class UserWithAssertedAttributes
+  def initialize(user)
+    @user = user
+  end
+
+  def call(attrs)
+    self.asserted_attributes = attrs
+  end
+
+  delegate :email, :uid, to: :user
+
+  attr_reader :asserted_attributes
```

Comment:
this is a very SAML-y term http://docs.oracle.com/cd/E27515_01/common/tutorials/attributes_saml_assertion.html